java/KinesisConnectors/src/main/java/com/amazonaws/services/msf/StreamingJob.java

```diff
@@ -44,39 +48,49 @@ private static Map<String, Properties> loadApplicationProperties(StreamExecution
         }
     }
 
-    private static KinesisStreamsSource<String> createKinesisSource(Properties inputProperties) {
+    private static <T> KinesisStreamsSource<T> createKinesisSource(Properties inputProperties, final DeserializationSchema<T> deserializationSchema) {
         final String inputStreamArn = inputProperties.getProperty("stream.arn");
-        return KinesisStreamsSource.<String>builder()
+        return KinesisStreamsSource.<T>builder()
                 .setStreamArn(inputStreamArn)
                 .setSourceConfig(Configuration.fromMap(Maps.fromProperties(inputProperties)))
-                .setDeserializationSchema(new SimpleStringSchema())
+                .setDeserializationSchema(deserializationSchema)
                 .build();
     }
 
-    private static KinesisStreamsSink<String> createKinesisSink(Properties outputProperties) {
+    private static <T> KinesisStreamsSink<T> createKinesisSink(Properties outputProperties, final SerializationSchema<T> serializationSchema) {
         final String outputStreamArn = outputProperties.getProperty("stream.arn");
-        return KinesisStreamsSink.<String>builder()
+        return KinesisStreamsSink.<T>builder()
                 .setStreamArn(outputStreamArn)
                 .setKinesisClientProperties(outputProperties)
-                .setSerializationSchema(new SimpleStringSchema())
+                .setSerializationSchema(serializationSchema)
                 .setPartitionKeyGenerator(element -> String.valueOf(element.hashCode()))
                 .build();
     }
 
+    private static <T> void createApplication(Class<T> classLiteral, final StreamExecutionEnvironment env, final Map<String, Properties> applicationProperties, final DeserializationSchema<T> deserializationSchema, final SerializationSchema<T> serializationSchema) {
+        KinesisStreamsSource<T> source = createKinesisSource(applicationProperties.get("InputStream0"), deserializationSchema);
+        KinesisStreamsSink<T> sink = createKinesisSink(applicationProperties.get("OutputStream0"), serializationSchema);
+
+        DataStream<T> input = env.fromSource(source,
+                WatermarkStrategy.noWatermarks(),
+                "Kinesis source",
+                TypeInformation.of(classLiteral));
+
+        input.sinkTo(sink);
```

Comment:
Maybe let's add some simple modification now that we have the class? 

nit: Would prefer passing in `TypeInformation` rather than class literal.